Updated Row API doc for RTL in textDirection

Pragya007 opened the pull request:
## Description

Updated Row API doc for RTL layout in textDirection.

## Related Issues

Fixes [#58019](https://github.com/flutter/flutter/issues/58019).


## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

AyushBherwani1998 commented on packages/flutter/lib/src/widgets/basic.dart:
```diff
@@ -3912,7 +3912,12 @@ class Flex extends MultiChildRenderObjectWidget {
   ///
   /// Defaults to the ambient [Directionality].
-  ///
+  /// 
+  /// If `textDirection` is [TextDirection.rtl], then the direction in which 
+  /// text flows starts from right to left. Otherwise, if `textDirection` is 
+  /// [TextDirection.ltr], then the direction in which text flows starts from 
+  /// left to right.
+  /// 
```
```suggestion
  ///
  /// If `textDirection` is [TextDirection.rtl], then the direction in which
  /// text flows starts from right to left. Otherwise, if `textDirection` is
  /// [TextDirection.ltr], then the direction in which text flows starts from
  /// left to right.
  ///
```

Pragya007 replied:
Will fix it.....Thank you..

HansMuller replied:
Although this is a correct statement about `textDirection` in general, it's not really help for understanding how the `Row` lays out its children.  If you rephrased the point (in terms of the row's children) and explained the implications of the start and end parameters, that would help.

#58019 was really a request to enhance the `Row` _class_ API doc, by mentioning the support for RTL layout a little earlier.

Pragya007 replied:
Sure. @HansMuller

goderbauer commented:
/cc @HansMuller Is this the clarification you had in mind when you filed https://github.com/flutter/flutter/issues/58019?

Pragya007 commented:
I want to add png image in the doc, how can I do this?

tvolkert committed:
Mark smoke_catalina_start_up_ios as flaky (#61737)

https://github.com/flutter/flutter/issues/61736

jmagman committed:
Show device diagnostic messages in doctor (#61585)

godofredoc committed:
Use device-finder instead of dev_finder is flutter code. (#61646)

engine-flutter-autoroll committed:
Roll Engine from d3b81f19fc98 to 1493883bf0fc (30 revisions) (#61658)

CareF committed:
cleanup macrobenchmark duplicate files (#61621)

* cleanup macrobenchmark duplicate files

engine-flutter-autoroll committed:
Roll Engine from 1493883bf0fc to d32782863249 (18 revisions) (#61743)

Commit:
Add no-op callbacks to platform view gesture recognizer when necessary (#61671)

After #31935, Some one sequence gesture recognizers requires at least one callback to be able to compete in the arena. This PR adds the a no-op callback in the gesture recognizer in the platform view when the gesture recognizer does not have any callbacks. This way, all the gesture recognizers in the platform view can compete in the arena.

iJimmyWei committed:
Fix overflow issue caused by a long title/subtitle for the vertical stepper (#61623)

chunhtai committed:
add static method to create material hero controller (#61662)

CareF committed:
benchmarkLive: a new `LiveTestWidgetsFlutterBindingFramePolicy` for benchmark on device (#61388)

* add benchmarkLive flag and tests

* update handlePointerEventRecord doc

* using e2e 0.6.1

christopherfujino committed:
add new docker_attach.sh script and update README (#61363)

jason-simmons committed:
Track used Android pointer IDs and reuse IDs after a pointer is released (#61668)

goderbauer committed:
Remove TODO arround deprecation (#61672)

engine-flutter-autoroll committed:
dcc9a4048 Roll Fuchsia Mac SDK from UAgiE... to fu1KK... (flutter/engine#19840) (#61759)

Commit:
[flutter_tools] revert dart format changes (#61760)

jmagman committed:
Inject iproxy into context (#61762)

helin24 committed:
Skip printing rendered error text in machine mode (#61684)

a14n committed:
restore nullability of FlutterError.onError (#61844)

chunhtai committed:
Revert "fix intrinsic height and width for widget span (#61485)" (#61876)

This reverts commit fc09119d0af589ade9c7e0609777a46ed773f5d7.

CareF committed:
LiveTestWidgetsFlutterBinding support for non-touch event (#61872)

Commit:
[flutter_tools] Add plumbing for widget cache (#61766)

To support #61407 , the tool needs to check if a single widget reload is feasible, and then conditionally perform a fast reassemble.

To accomplish this, the FlutterDevice class will have a WidgetCache injected. This will eventually contain the logic for parsing the invalidated dart script. Concurrent with the devFS update, the widget cache will be updated/checked if a single widget reload is feasible. If so, an expression evaluation with the target type is performed and the success is communicated through the devFS result. An integration test which demonstrates that this works is already present in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/hot_reload_test.dart#L86

Finally, when actually performing the reassemble the tool simply checks if this flag has been set and calls the alternative reassemble method.

Cleanups:

Remove modules, as this is unused now.

CareF committed:
Revert "LiveTestWidgetsFlutterBinding support for non-touch event (#61872)" (#61900)

This reverts commit aa4b4d35e1187483fec4ec71fc278e724257cd77.

bkonyi committed:
Reland "Add support for Dart Development Service (DDS) in Flutter Tools (#61276)" (#61882)

This reverts commit 38fe887342f6c6bf40dd2c25442b892644e11a39.

bkonyi committed:
Revert "Reland "Add support for Dart Development Service (DDS) in Flutter Tools (#61276)" (#61882)" (#61909)

This reverts commit 895b7ef6faf4e9c6ad641ad556855ff38fbd04bb.

darrenaustin committed:
Ignore .idea/* files in the analysis bot. (#61215)

xu-baolin committed:
InkDecoration do not paint if it's part of the tree doesn't get painted (#61216)

a14n committed:
migrate scheduler to nullsafety (#61570)

jason-simmons committed:
Disable the deferred loading variant of gen_l10n_test (#61912)

Commit:
Add more tips for androidLicensesUnknown (#61312)

Commit:
[flutter_tools] turn off fuchsia support by default (#61634)

DetectiveAI4869 committed:
[flutter_tools] Concise dart2js commands outputKernel & outputJavaScript (#61881)

CareF committed:
Reland "LiveTestWidgetsFlutterBinding support for non-touch event" (#61901)

guidezpl committed:
Implement customizable cursor height (#61714)

CareF committed:
remove frame policy test flaky (#61921)

engine-flutter-autoroll committed:
Roll Engine from dcc9a4048da8 to 5a73d4dc2503 (36 revisions) (#61944)

chunhtai committed:
Revert "InkDecoration do not paint if it's part of the tree doesn't get painted (#61216)" (#61963)

This reverts commit fdf87edd2fadb3ba3219d23792f0210384533e1b.

Piinks commented on packages/flutter/lib/src/widgets/basic.dart:
````diff
@@ -4144,6 +4149,31 @@ class Flex extends MultiChildRenderObjectWidget {
+/// )
+/// ```
+///
+/// ![](https://flutter.github.io/assets-for-api-docs/assets/widgets/row_textDirection.png)
````
This link is broken. I don't think this asset actually exists in https://github.com/flutter/assets-for-api-docs?

Pragya007 replied:
@Piinks  Yes it is but I have also created PR for that too, [#115](https://github.com/flutter/assets-for-api-docs/pull/115)

bkonyi committed:
Reland "Add support for Dart Development Service (DDS) in Flutter Tools (#61276)" (#61975)

This reverts commit 4867f5931ff87a88fd3e89e7cf1b3cd9bd427670.

bkonyi committed:
Revert "Reland "Add support for Dart Development Service (DDS) in Flutter Tools (#61276)" (#61975)" (#61993)

This reverts commit f7a1c87ffdb9476c25cda487cb7232617be31b33.

Commit:
Revert "Roll Engine from dcc9a4048da8 to 5a73d4dc2503 (36 revisions) (#61944)" (#62050)

This reverts commit d9252f95f9b85e72c0c677f5d87db3aeab1835c2.

jmagman committed:
Mark flutter_gallery_ios__transition_perf flaky during investigation (#62082)

jmagman committed:
Only try to get build settings if the Xcode project exists (#61979)

christopherfujino committed:
update ios-deploy (#61983)

Commit:
[flutter_tools] macOS cleanups, attach to log reader in release mode (#61913)

YazeedAlKhalaf committed:
✒ Spell Check All `.md` Files Related to Flutter 💙 (#61564)

* 🐛 Fix Spelling Issues in Main README.md

* 🐛 Fix spelling issues in dev README.md

* 🐛 Fix spelling issues in complex_layout README.md

* 🐛 Fix spelling issues in macrobenchmarks README.md

* 🐛 Fix spelling issues in platform_views_layout README.md

* 🐛 Fix spelling issues in test_Apps/stocks README.md

* 🐛 Fix spelling issues in bots README.md

* ✒ Spell Check dev/ci README.md

* ✒ Spell Check dev/ci/docker_linux README.md

* ✒ Spell Check dev/devicelab README.md

* ✒ Spell Check dev/docs README.md

* ✒ Spell Check dev/snippets README.md

* ✒ Spell Check dev/snippets/config/templates README.md

* ✒ Spell Check dev/tools/gen_keycodes README.md

* ✒ Spell Check dev/tools/vitool README.md

* ✒ Spell Check examples/catalog README.md

* ✒ Spell Check examples/flutter_view README.md

* ✒ Spell Check examples/image_list README.md

* ✒ Spell Check examples/layers README.md

* ✒ Spell Check examples/platform_channel README.md

* ✒ Spell Check examples/platform_channel_swift README.md

* ✒ Spell Check examples/platform_view README.md

* ✒ Spell Check packages/_flutter_web_build_script README.md

* ✒ Spell Check packages/flutter_localizations README.md

* ✒ Spell Check packages/flutter_tools README.md

* ✒ Spell Check CODE_OF_CONDUCT.md

* ✒ Spell Check dev/integration_test/android_splash_screens/splash_Screen_load_rotate README.md

* ✒ Spell Check dev/integration_test/android_views README.md

* ✒ Spell Check dev/integration_tests/flutter_driver_screenshot_test README.md

* ✒ Spell Check dev/integration_tests/flutter_gallery README.md

* ✒ Spell Check dev/integration_tests/gradle_deprecated_settings README.md

* ✒ Spell Check dev/integration_tests/ios_add2app_life_cycle README.md

* ✒ Spell Check dev/integration_tests/ios_host_app README.md

* ✒ Spell Check dev/integration_tests/ios_platform_view_tests README.md

* ✒ Spell Check dev/automated_tests/flutter_test README.md

* ✒ Spell Check .github/PULL_REQUEST_TEMPLATE.md

* ✒ Spell Check .hithub/ISSUE_TEMPLATE/ACTIVATION.md

* ✒ Spell Check .github/ISSUE_TEMPLATE/BUG.md

* ✒ Spell Check .github/ISSUE_TEMPLATE/feature_request.md

* ✒ Spell Check .github/ISSUE_TEMPLATE/performance_others.md

* ✒ Spell Check .github/ISSUE_TEMPLATE/performance_speed.md

* ✒ Spell Check packages/flutter_tools/doc/daemon.md

* ✒ Spell Check packages/flutter_tools/fuchsia_enrtypoint_shim/README.md

* ✒ Minimize line to 80 columns

* ✒ Minimize line to 80 columns

* ✒ Fix Typo

* ✒ Chnaged numbers to 1 for testing purposes

* ✒ Changed numbers to 1 for testing purposes

* ❌ Remove 'a' which is a typo

* ✒ Change a sentence to be better

* ✒ Remove 'a' which is a typo

* ✒ Fix small issue

* ✒ Fix small typo

* ✒ Fix some typos

* ❌ Remove trailing space

* ❌ Remove trailing space

* 🐛 Fix small typo

* ✒ Fix Typo

* 🐛 Fix small bug

Chaoyang0201 committed:
Show correct errors when plugins yaml forgot the 'flutter.plugins.pla… (#61338)

davidmartos96 committed:
Use the correct Transform in the WidgetInspector overlay (#59566) (#60990)

guidezpl committed:
Add per thumb Range Slider semantics (#61439)

tvolkert committed:
Update Actions.invoke to return the result of invoking the action. (#62012)

The boolean value as to whether the action was found and enabled
is less useful than just returning the result of invoking the action
itself.

tvolkert committed:
Minor doc updates (#62008)

CareF committed:
implement handlePointerEventRecord for LiveWidgetController (#61266)

Pragya007 committed:
Updated instructions of 'flutter help attach' (#61569)

gspencergoog committed:
Find the target platform more accurately on the web. (#61587)

Commit:
Reland "Branch out android platform view perf testing #60238" 2nd time (#61748)

Commit:
[flutter_tools] clean plugin dependencies file (#62061)

Clean even more, in case the plugin dependency update logic is wrong-o.

Commit:
[flutter_tools] check in script for generating per library unit coverage (#61996)

Allows generating a per-library coverage summary like https://gist.github.com/jonahwilliams/f298381c3fb9f472b2dfe54b82a20a88

chunhtai committed:
remove usage to applyTextScaleFactorToWidgetSpan (#61636)

arafaysaleem committed:
Allowed specifying reverseTransitionDuration in PageRouteBuilder class (#61752)

Commit:
Test dynamic surface switch (#61918)

gspencergoog committed:
Fix `ExcludeFocus` so it won't refocus a sibling of the focused node. (#61756)

This changes FocusNode.descendantsAreFocusable so that it doesn't allow the enclosing scope to re-focus a node that is a descendant of the node with descendantsAreFocusable set to false.

Because of the order in which the internal state for descendantsAreFocusable was being set, setting it to false was causing a sibling node to be focused when descendantsAreFocusable of the parent was set to false, even though it shouldn't have been focusable, because the enclosing scope would search for a candidate to be focused before the internal state was set to false.

Instead of looping over the children and telling them all to unfocus (and select the previously focused node), this unfocuses the node that has descendantsAreFocusable set to false, with the disposition UnfocusDisposition.previouslyFocusedChild, so that its enclosing scope will look for a previously focused child that isn't part of the subtree being excluded.

This affects how the ExcludeFocus widget behaves when turning on exclude.

goderbauer committed:
Make it an error to break macros in docs (and fix existing broken macros) (#62071)

goderbauer committed:
Update viewport dimensions when scrollDirection changes (#61973)

CareF committed:
Add a E2E based performance test case (#61509)

LongCatIsLooong committed:
Check if a double is NaN before converting to it int (#61940)

arafaysaleem committed:
Improved docs for minLines in EditableText (#61796)

a14n committed:
migrate physics to nullsafety (#61941)

pedromassango committed:
Update basic.dart (#61961)

a14n committed:
Add template flutter.widgets.transitionRoute.reverseTransitionDuration (#62136)

chunhtai committed:
Revert "remove usage to applyTextScaleFactorToWidgetSpan (#61636)" (#62144)

This reverts commit c816fabd0af48a861afdeb78bb6ebe4b7cbd3707.

jason-simmons committed:
Manual engine roll to 4d720dd9e56ff5d008ed8cafff3a45676f403c23 (#62150)

jason-simmons committed:
Disable DartDev when running device lab tests (#62154)

This is a workaround for https://github.com/flutter/flutter/issues/62139
so that the engine roll pipeline can proceed.

jason-simmons committed:
Manual engine roll to 602c2d81fa166c77920c2b2fdc190a6f2f257985 (#62158)

jason-simmons committed:
Manual engine roll to 626244a72c5d53cc6d00c840987f9059faed511a (#62163)

Disabled web_benchmarks_canvaskit in the bot scripts because it was
failing in the Cirrus environment.

zanderso committed:
Pass --disable-dart-dev in more places (#62217)

Commit:
[flutter_tools] deflake fastReassemble test (#62152)

AyushBherwani1998 committed:
[CONTRIBUTING.md] adds Contributing to Flutter article link (#61517)

zanderso committed:
Ensure analytics are sent before disabling (#62094)

Commit:
[flutter_tools] ensure x64 is built if target-platforms is not specified (#62096)

In the add-2-app case, users may not specify the android target platforms field, but this only defaults to arm builds. Add x64 to the default list.

Fixes #62095

Commit:
[flutter_tools] remove breakpoints from paused isolate on hot restart (#62069)

The embedder requires that the isolate is unpaused, because the runInView method requires interaction with dart engine APIs that are not thread-safe. These APIs must be run on the same thread that would be blocked by the pause. Simply unpausing is not sufficient, because this does not prevent the isolate from immediately hitting a breakpoint, for example if the breakpoint was placed in a loop or in a frequently called method. Instead, all breakpoints are first disabled and then the isolate resumed.

Piinks committed:
Remove unused FlutterErrorDetails subclasses (#61579)

speaking-in-code committed:
Make sure all isolates start during flutter driver tests. (#61841)

pedromassango committed:
Add minWidth and height parameter into FlatButton widget (#61664)

willlarche committed:
[Material] Updated icons and fonts (#61778)

tvolkert committed:
Add support for system alert sound (#62086)

* Add SystemSoundType.alert (supported in the engine in flutter/engine#19970)
* Play system alert sound when user tries to dismiss a non-dismissable modal
  barrier

tvolkert committed:
Fix error handling for errors with empty stack traces (#62224)

Fixes https://github.com/flutter/flutter/issues/62223

jmagman committed:
Detect exact device ID matches quickly (#62070)

TooYoungTooSimp committed:
[flutter_tools] Make flutter git upstream configurable (#61513)

a14n committed:
bump nullsafety deps (#61978)

zanderso committed:
Revert "bump nullsafety deps (#61978)" (#62234)

This reverts commit c46aa360b0dfd30895c85f698bc4588ab164dcd0.

Commit:
Revert "Make sure all isolates start during flutter driver tests. (#61841)" (#62239)

This reverts commit 5fa5701d72970a2a176d75377bca00539593e98f.

tvolkert committed:
Print date/time for each step taken in docs.sh (#62248)

This will allow us to see in the logs how long each step takes

tvolkert committed:
Add --disable-dart-dev to docs.sh (#62249)

https://github.com/flutter/flutter/issues/62206

tvolkert committed:
Rather than timing out dashing, run it in background (#62251)

https://github.com/flutter/flutter/issues/60646

tvolkert committed:
Sub-shard the docs shard (#62281)

Separate out offline docs, docset generation, and docs deployment
each into dedicated shards.

https://github.com/flutter/flutter/issues/60646

tvolkert committed:
Fix doc generation scripts (#62340)

Commit:
roll dart/engine to f27729e97b5a00c3a8d8d49edc7b998fa755b97a (#62368)

tvolkert committed:
Use in-memory disk for docs shards (#62373)

https://github.com/flutter/flutter/issues/60646

tvolkert committed:
Temporarily disable docset generation (#62386)

https://github.com/flutter/flutter/issues/60646

tvolkert committed:
Add timeout to the devicelab runner logs (#62418)

monkeyswarm committed:
Adds an onMove callback on a DragTarget. (#60174)

chunhtai committed:
clean up hero controller scope (#60655)

jmagman committed:
Parse .flutter-plugins-dependencies for add-to-app iOS modules (#61269)

Pragya007 committed:
Added expandedHeight in flexibleSpace and flexibleSpaceBar (#61710)

goderbauer committed:
Restoration Framework (#60375)

* state restoration

* added example

* typos and analyzer

* whitespace

* more typos

* remove unnecessary import

* whitespace

* fix sample code

* tests for restorationmanager and restorationid

* ++

* typo

* tests for bucket, part1

* rename tests

* more tests

* finished tests for service layer

* remove wrong todo

* ++

* review comments

* tests for Unmanaged and regular scope

* RootRestorationScope tests

* typo

* whitespace

* testing framework

* tests for properties

* last set of tests

* analyzer

* typo

* dan review

* whitespace

* ++

* refactor finalizers

* ++

* ++

* dispose guard

* ++

* ++

* dan review

* add manager assert

* ++

* analyzer

* greg review

* fix typo

* Ian & John review

* ian review

* RestorationID -> String

* revert comment

* Make primitives non-nullable in prep for NNBD

Commit:
[flutter_tools] fix exit label for batchfile (#62272)

move the final_exit label to the correct script location for the flutter/dart entrypoints.

Commit:
[flutter_tools] update device discovery constructors (#61640)

google3 callsites are updated, remove global lookup from constructor parameters.

liyuqian committed:
Add a test for 3D transform with needsCompositing (#61779)

songyiYu committed:
Add LayoutBuilder example (#61948)

DetectiveAI4869 committed:
[flutter_tools] Unified analyze error log format (#61146)

* Unified analyze error log format

* Fix review issues

LongCatIsLooong committed:
Autofill save (#58731)

chunhtai committed:
makes PopupMenuitem merge the semantics of its child (#62062)

rrousselGit committed:
Update the documentation of GlobalKey to explain why it is expensive (#62083)

jmagman committed:
Allow hyphens in iOS UDIDs (#62225)

jmagman committed:
Mark smoke_catalina_start_up_ios as not flaky (#62085)

Commit:
remove iOS folder (#61561)

tvolkert committed:
Fix some broken doc references (#62317)

https://github.com/flutter/flutter/issues/31931

tvolkert committed:
Export Material curves in the Material lib (#62421)

(split out from https://github.com/flutter/flutter/pull/62317)

Piinks committed:
Fix crossSize for baseline (#62074)

darrenaustin committed:
Updated the About dialog to use the new button API. (#62075)

darrenaustin committed:
Updated references to the old button classes to the new ones in comments and documentation for classes in the material library. (#62092)

HansMuller committed:
Replace mistaken dartdoc [Foo..bar] references with [Foo.bar] (#62338)

CareF committed:
Add WidgetController.scrollUntilVisible (#62097)

darrenaustin committed:
Updated references to the old button classes to the new ones in comments and documentation for classes in the widgets, services and rendering libraries. (#62099)

Pragya007 committed:
Updated 'CustomClipper.shouldReclip' doc (#62101)

CareF committed:
Simplify macrobencharks/picture_cache (#62219)

liyuqian committed:
Add Crane test to the new Gallery app (#62162)

darrenaustin committed:
Updated references to the old button classes to the new ones in comments and documentation for tests in the material and cupertino libraries. (#62164)

Commit:
Revert "Add Crane test to the new Gallery app (#62162)" (#62430)

This reverts commit 1ffdfa81d6ba4258ed586a1b0905de74e3961645.

Commit:
Revert "Export Material curves in the Material lib (#62421)" (#62431)

This reverts commit 80237ef8d66d2ee0988c754f7686a290a7dfa3d5.

goderbauer committed:
Update the FlutterLogo colors and gradients (#62052)

a14n committed:
migrate gestures to nullsafety (#62157)

rami-a committed:
Ensure banner reads banner theme's leading padding property (#62195)

goderbauer committed:
Fix "unresolved doc reference" in rendering layer (#62443)

Alexays committed:
Fix README typo (#62218)

ruskakimov committed:
Fix typos in implicit_animations.dart (#62287)

darrenaustin committed:
Updated the framework tests to use the new button API. (#62220)

goderbauer committed:
Bump new gallery version (#62439)

engine-flutter-autoroll committed:
Roll Engine from f27729e97b5a to 2d8a00e21d8c (9 revisions) (#62438)

goderbauer committed:
Ban unresolved dartdoc directives from HTML output (#62167)

goderbauer committed:
Make dartdoc warnings fatal (& cleanup of warnings) (#62367)

CareF committed:
bug fix (#62349)

Commit:
[flutter_tools] ensure desktop builds can run without flutter_tools' package_config (#62426)

chingjun committed:
Wrap launching devtools in DevtoolsLauncher (#62364)

goderbauer committed:
Fix "unresolved doc reference" in widgets layer - part 1 (#62452)

goderbauer committed:
Revert "Revert "Export Material curves in the Material lib (#62421)" (#62431)" (#62435)

engine-flutter-autoroll committed:
Roll Engine from 2d8a00e21d8c to 1ba310017c58 (6 revisions) (#62458)

engine-flutter-autoroll committed:
Roll Engine from 1ba310017c58 to a9910e409c02 (2 revisions) (#62469)

CareF committed:
WidgetController.pump use optional duration (#62091)

rami-a committed:
[Material] Add support for high contrast theming to Material apps (#62337)

CareF committed:
Move key event and semantics related method from WidgetTester to WidgetController (#62362)

Commit:
[flutter_tools] add flutterRoot field to JSON machine output (#62053)

justinmc committed:
SelectableText handles after Select All (#62072)

bkonyi committed:
Reland "Add support for Dart Development Service (DDS) in Flutter Tools (#61276)" (#62147)

This reverts commit adc9dde3ba8563eebb824feb689f95eb947ab745.

- Fixed issue where `FallbackDiscovery` would hold on to a `VmService` when launching on iOS devices, causing DDS to fail to start
- Fixed `flutter drive` case where DDS is already running in another flutter_tools instance

tvolkert committed:
Fix Border.symmetric: phase 3 (#62471)

This removes the (now unused) `invertMeaningOfVerticalAndHorizontal`
flag.

Fixes https://github.com/flutter/flutter/issues/61470

goderbauer committed:
Fix 'unresolved doc reference' for material A-D (#62473)

goderbauer committed:
Fix todo in restoration framework (#62500)

liyuqian committed:
Add purge-persistent-cache option to run and drive (#62221)

Fixes https://github.com/flutter/flutter/issues/39526

Needs https://github.com/flutter/engine/pull/20013 to land and roll first.

bkonyi committed:
Temporarily disable Dart Development Service in flutter_tools (#62508)

Devtools doesn't currently support connections to DDS. Disable DDS
temporarily while a solution is worked on. See https://github.com/flutter/flutter/issues/62507

a14n committed:
migrate animation to nullsafety (#62485)

* migrate animation to nullsafety

* nullable tween

* fix generic type issue

* address review comment

dkwingsmt committed:
Refactor gen_keycode: Split generators (#61916)

* Split gen_keycode to make maintenance easier.

HansMuller committed:
Added OverflowBar widget (#62350)

robert-ancell committed:
Set the application-id in the Linux shell (#60053)

tvolkert committed:
Remove devicelab timeouts (#62511)

goderbauer committed:
Fix more unresolved doc references in material (#62509)

goderbauer committed:
fix unresolved doc references in widgets - part 2 (#62499)

dkwingsmt committed:
Add tests for platform views' hover behavior  (#61667)

flar committed:
ignore .idea directory in dart analyzer and analyzer benchmark output (#55235)

Commit:
[flutter_tools] switch cache mode to reload (#62076)

JuYeong0413 committed:
Update image docs that Image.file() cashes file (#62301)

AyushBherwani1998 committed:
replaces deprecated api usage for android (#62515)

mehmetf committed:
Expose certain functions used when rolling Flutter (#62529)

goderbauer committed:
Fix even more unresolved doc references in material (#62520)

* Fix even more unresolved doc references in material

* ++

* ++

tvolkert committed:
Add comment to developers about adding new system sound types (#62552)

flutter/flutter#62143

xrr2016 committed:
Add padding around flutter api docs search input (#44980)

Commit:
[flutter_tools] delete test that will start failing (#62580)

xu-baolin committed:
Ignore unmounted parent when _debugVerifyGlobalKeyReservation for GlobalKey check (#62323)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

* Ignore unmounted parent when _debugVerifyGlobalKeyReservation (#62055)

Commit:
[flutter_tools] deflake integration tests by running single isolate (#62534)

There are some occasional difficult to repro flakes on the tool integration tests. Try running only a single isolate to simplify the environment a bit.

goderbauer committed:
Random unresolved doc reference fixes I missed before (#62582)

CareF committed:
Update framePolicy docs (#61773)

gspencergoog committed:
Fix doc warnings for focus code (#62602)

angjieli committed:
Allow Flutter Web Driver testing in debug mode. (#62577)

Commit:
[flutter_tools] fix formatting regression in analyze and add full coverage (#62593)

Commit:
[flutter_tools] update pubspec .yaml message should not be displayed when creating a new plugin (#62599)

* update pubspec message should not be displayed when creating a new plugin

* formatting

* revikew

* review

* cleanup

goderbauer committed:
After cleaning up 812 warnings: Make warnings fatal in dartdocs (#62553)

Commit:
[flutter_tools] add snapshot analysis dependency (#62590)

digiter committed:
[devicelab] Mark flutter_gallery_ios32__start_up as flaky. (#62646)

The test device is causing failure of this test, we will need a couple days to physically access the device and fix the issue. So mark it as flaky to unblock the tree.

Bug: https://github.com/flutter/flutter/issues/62633

Commit:
Fix SemanticsNode.rect position for nested scrollables with useTwoPaneSemantics (#62359)

keyonghan committed:
remove flaky flag (#62348)

jason-simmons committed:
Make the text field macrobenchmark work in debug mode (#62523)

liyuqian committed:
Reland "Add Crane test to the new Gallery app (#62162)" (#62544)

goderbauer committed:
Alphabetize warning list (#62611)

chunhtai committed:
Reland "remove usage to applyTextScaleFactorToWidgetSpan" (#62587)

This reverts commit 561735ad89a14c63597f9acb77d447c7e0ed4fec.

peterdjlee committed:
Implement size analyzer to unzip & parse APK and AOT size snapshot to generate analysis json (#62495)

* Implement size analyzer to unzip & parse APK and AOT size snapshot to generate analysis json

engine-flutter-autoroll committed:
Roll Engine from a9910e409c02 to 280bbfc763cf (49 revisions) (#62647)

* 4521162a0 Roll Skia from 39d1c1ebb607 to 3a608e5bc9d5 (1 revision) (flutter/engine#20101)

* b265bd2ba Roll Skia from 3a608e5bc9d5 to 620bfa3fffba (1 revision) (flutter/engine#20102)

* 36c5f6081 Roll Skia from 620bfa3fffba to 4866d0ad5f3c (1 revision) (flutter/engine#20104)

* 06fef5e36 Enable lazy-async-stacks by-default in all modes (2) (flutter/engine#19270)

* d71b11f94 Roll Fuchsia Linux SDK from ETPOTPepP... to ROYgzKMaF... (flutter/engine#20105)

* 42e4ebf06 Roll Skia from 4866d0ad5f3c to c34efe0da102 (1 revision) (flutter/engine#20108)

* 3a65b1a65 Roll Skia from c34efe0da102 to 8c7ecc1c06f4 (1 revision) (flutter/engine#20109)

* ad99f5e50 Roll Skia from 8c7ecc1c06f4 to 4f587337c306 (1 revision) (flutter/engine#20110)

* f0cc38fdb [web] Set correct defaults for text in canvas (flutter/engine#20067)

* 02f9ed908 Roll Fuchsia Mac SDK from 5dM55hp8B... to hzo88TZzN... (flutter/engine#20113)

* 34389f505 Roll Skia from 4f587337c306 to 96d6c6f04dcb (4 revisions) (flutter/engine#20114)

* 49a40faba Enabled linting on engine.cc (flutter/engine#19981)

* b955e15ee Manual roll of Dart from 24c7666def...40fd1c456e (flutter/engine#20092)

* 145ef60b9 Remove confusing logs (flutter/engine#20121)

* 8fc3926a8 Roll Skia from 96d6c6f04dcb to 57d859edd3c4 (16 revisions) (flutter/engine#20126)

* ec9e17c5c Roll zlib (flutter/engine#20119)

* f288fe563 [web] Enable canvas measurement by default (flutter/engine#19924)

* 846420886 Add missing MouseCursorPlugin destroy call (flutter/engine#19968)

* 5498add19 Roll Skia from 57d859edd3c4 to 994ce8cf2300 (1 revision) (flutter/engine#20129)

* 9398bc4b1 Roll Skia from 994ce8cf2300 to 398c654ce7be (2 revisions) (flutter/engine#20133)

* d3bc43e74 Roll Fuchsia Linux SDK from ROYgzKMaF... to d4pESQYnB... (flutter/engine#20132)

* 406891841 Manual roll of Dart 40fd1c456e...7e8348f4ce (flutter/engine#20125)

* 4c49e0b19 Manual roll of Dart cb6ed67a73...7e8348f4ce (flutter/engine#20135)

* 4a3688e47 Roll Skia from 398c654ce7be to a4bbc9d8ec4f (1 revision) (flutter/engine#20136)

* b3c6fd311 Roll Fuchsia Mac SDK from hzo88TZzN... to 3XwiR_wVO... (flutter/engine#20137)

* adb59861a Manual roll of Dart 03e4737f31...cb6ed67a73 (flutter/engine#20138)

* 941c442b8 Add ALERT SoundType enum value (flutter/engine#20139)

* 19368efcc Fix dartdocs of dart:ui (flutter/engine#20140)

* 9dd3d2ea7 Roll Skia from a4bbc9d8ec4f to 94cefeff50d2 (1 revision) (flutter/engine#20141)

* 0fdb14143 Roll Dart SDK from 03e4737f3115 to 59600f2b46ef (54 revisions) (flutter/engine#20143)

* 97d6ee26a Roll Skia from 94cefeff50d2 to 5ba6534884d9 (2 revisions) (flutter/engine#20146)

* da3d4958b Roll Dart SDK from 59600f2b46ef to 04f4272546af (5 revisions) (flutter/engine#20147)

* 8229df872 Roll Skia from 5ba6534884d9 to e393c61a1563 (1 revision) (flutter/engine#20148)

* 8e1d48eec Migrate some Dart_WeakPersistentHandle uses to Dart_FinalizableHandle (flutter/engine#20107)

* 27b61e7ac Roll ANGLE to pick up warning fixes (flutter/engine#19105)

* 146d50490 Roll Skia from e393c61a1563 to 3136789972ea (5 revisions) (flutter/engine#20150)

* 841b3917c Roll Dart SDK from 04f4272546af to e87cb96bb89c (7 revisions) (flutter/engine#20152)

* e9334c9a4 Roll Skia from 3136789972ea to 5f2b2d6dc691 (5 revisions) (flutter/engine#20153)

* f2b02d8fd [iOS] Fixes text input plugin crash (flutter/engine#20127)

* 3ed5893ac Roll Skia from 5f2b2d6dc691 to 8cc118dce813 (2 revisions) (flutter/engine#20154)

* 7c5162e0a Roll Fuchsia Mac SDK from 3XwiR_wVO... to T2xc0OuiK... (flutter/engine#20155)

* e23e4775a Lint fixes for fml, tools subdirs (flutter/engine#19990)

* f620eacd7 Roll Dart SDK from e87cb96bb89c to bd528bfbd69d (8 revisions) (flutter/engine#20158)

* e1c9673bb Fix targets in build_fuchsia_artifacts (flutter/engine#19794)

* c134e1658 add information collection for safari bots (flutter/engine#20123)

* ee4d50cd0 Revert "Enable lazy-async-stacks by-default in all modes (2) (#19270)" (flutter/engine#20165)

* 357b155ef Roll Fuchsia Linux SDK from d4pESQYnB... to d_5wDVmBd... (flutter/engine#20161)

* a6cd3ebc6 Fix lint errors in lib/ui (flutter/engine#19988)

* 280bbfc76 This makes the lint script use multiprocessing to speed it up. (flutter/engine#19987)

helin24 committed:
Add daemon handler to start devtools (#62608)

a14n committed:
migrate services to nullsafety (#62513)

NNBD migration for services

Commit:
remove flutter attach devicelab test (#62704)

The flutter tool already has an attach integration test: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/flutter_attach_test.dart

The devicelab test frequently fails with difficult to debug issues and is not worth the maintenance cost.

engine-flutter-autoroll committed:
Roll Engine from 280bbfc763cf to 083282e33bbb (22 revisions) (#62768)

* 7f5d044f3 Wait before switching surfaces (flutter/engine#20100)

* 5513273d0 Reland: Avoid a copy in EncodeImage (flutter/engine#20003)

* 1efdd9538 Roll Dart SDK from bd528bfbd69d to ea6bde577d1c (19 revisions) (flutter/engine#20172)

* 3b0e697db Roll Skia from 8cc118dce813 to c3794dd52778 (27 revisions) (flutter/engine#20173)

* cb1a37401 Roll Fuchsia Mac SDK from T2xc0OuiK... to i0zTcQ8Qb... (flutter/engine#20175)

* ed36b1aa2 Roll Skia from c3794dd52778 to 2d01ed94605a (10 revisions) (flutter/engine#20179)

* fcc1eaf8f Fix iOS Keyboard stuck as UIKeyboardTypeNamePhonePad (flutter/engine#20181)

* 9c6837c2b Roll Skia from 2d01ed94605a to 7225788b9070 (6 revisions) (flutter/engine#20183)

* 13e993ed1 Fix Typos (flutter/engine#19691)

* 98cfd1db7 Move platform specific information to `PlatformConfiguration` class (flutter/engine#19652)

* 22fb58b77 update nullability of drawAtlas methods and flesh out docs (flutter/engine#20176)

* bcc43dfb6 Roll Dart SDK from ea6bde577d1c to 033a81d924b9 (23 revisions) (flutter/engine#20186)

* cb4bb93d9 [web] increase number of shards. sync engine web tests same as flutter repo (flutter/engine#20164)

* d986b8d60 Enable linting in several files (flutter/engine#20134)

* 7dd092dd3 Enable more linting (flutter/engine#20187)

* 3cc86ac13 Roll Dart SDK from 033a81d924b9 to ad5bcf16f1c8 (9 revisions) (flutter/engine#20191)

* 5ca8a2a0e Roll Dart SDK from ad5bcf16f1c8 to d169af6f7d8f (1 revision) (flutter/engine#20192)

* 4de0c048d Roll Dart SDK from d169af6f7d8f to 7e6c55e3aaf5 (1 revision) (flutter/engine#20196)

* 908fe012d Fix navigation message relay. (flutter/engine#20193)

* f1b3b6981 Roll Dart SDK from 7e6c55e3aaf5 to 365525432a70 (2 revisions) (flutter/engine#20197)

* 8fbdd3f1b Fix parameter names

* 083282e33 Fix Implments typo

jmagman committed:
Add integration test for plugins without ios directories (#62630)

guidezpl committed:
Cache floating cursor location on start event instead of update (#62415)

helin24 committed:
Return devtools serve response instead of printing (#62702)

Commit:
[versions] update to the latest sdk versions (#62687)

mehmetf committed:
Remove mistakenly added DartDevelopmentService dependency (#62792)

engine-flutter-autoroll committed:
f86b83010 Roll Dart SDK from 365525432a70 to 2d98d25e71c3 (1 revision) (flutter/engine#20198) (#62770)

goderbauer committed:
Make dartdoc warnings fatal for flutter_test (#62614)

Piinks commented on packages/flutter/lib/src/widgets/basic.dart:
```diff
@@ -4144,6 +4149,31 @@ class Flex extends MultiChildRenderObjectWidget {
 ///
 /// ![](https://flutter.github.io/assets-for-api-docs/assets/widgets/row_fixed.png)
 ///
+/// Now suppose you want to render children in different direction, this can be
+/// done by setting [textDirection] property.
+/// [TextDirection.ltr] is the default rendering of children of row so the first
+/// child is the `start` of row i.e. left and and moving to right last child is
+/// the `end` of row i.e. right. But, if you want to render it in opposite
+/// direction i.e. right to left, then [textDirection] can be set to
+/// [TextDirection.rtl] and this will render children starting from right to
+/// left. So the first child will become `end` of row starting from right and
+/// moving to left the last child is the `start` of row.
```
```suggestion
/// The [textDirection] property controls the direction that children are rendered in.
/// [TextDirection.ltr] is the default [textDirection] of [Row] children, so the first
/// child is rendered at the `start` of the [Row], to the left, with subsequent children
/// following to the right. If you want to order children in the opposite
/// direction (right to left), then [textDirection] can be set to
/// [TextDirection.rtl]. This is shown in the example below.
```

Piinks commented on packages/flutter/lib/src/widgets/basic.dart:
```diff
@@ -3913,6 +3913,11 @@ class Flex extends MultiChildRenderObjectWidget {
   ///
   /// Defaults to the ambient [Directionality].
   ///
+  /// If `textDirection` is [TextDirection.rtl], then the direction in which
```
```suggestion
  /// If [textDirection] is [TextDirection.rtl], then the direction in which
```

Piinks commented on packages/flutter/lib/src/widgets/basic.dart:
```diff
@@ -3913,6 +3913,11 @@ class Flex extends MultiChildRenderObjectWidget {
   /// Defaults to the ambient [Directionality].
   ///
+  /// If `textDirection` is [TextDirection.rtl], then the direction in which
+  /// text flows starts from right to left. Otherwise, if `textDirection` is
```
```suggestion
  /// text flows starts from right to left. Otherwise, if [textDirection] is
```

Piinks reviewed (commented):
The diagram you made has landed in the assets repo. 🎉 
There are just a few adjustments here. :)

jakemac53 committed:
update to use more null safe packages (#62525)

Updates all the dependencies used in the most basic test and flutter_test isolates.

Commit:
[flutter_tools] add --analyze-size flag (#62697)

engine-flutter-autoroll committed:
Roll Engine from f86b83010d6c to 02d71d60effa (6 revisions) (#62804)

* 8a0e30b36 Roll Dart SDK from 2d98d25e71c3 to c59cdee365b9 (1 revision) (flutter/engine#20200)

* 0519052c8 Roll Dart SDK from c59cdee365b9 to ad9c73cb3270 (1 revision) (flutter/engine#20201)

* fc5b7779d Roll Skia from 7225788b9070 to 8e77631a80f0 (8 revisions) (flutter/engine#20184)

* cb8dfdb29 Roll Skia from 8e77631a80f0 to a6df9da28963 (17 revisions) (flutter/engine#20202)

* ea811fc77 Roll Dart SDK from ad9c73cb3270 to 0f0e04ec3afa (3 revisions) (flutter/engine#20204)

* 02d71d60e Use a single mask view to clip iOS platform view (flutter/engine#20050)

HansMuller committed:
Use OverflowBar instead of ButtonBar in MaterialBanner (#62703)

a14n committed:
text selection affinity can be null on web (#62778)

HansMuller committed:
Use OverflowBar instead of ButtonBar in TimePicker (#62601)

HansMuller committed:
Use OverflowBar instead of ButtonBar in DatePicker (#62686)

HansMuller committed:
Use OverflowBar instead of ButtonBar in DateRangePicker (#62695)

darrenaustin committed:
Fix test lifecycle clean up for screen size and semantics. (#62761)

engine-flutter-autoroll committed:
Roll Engine from 02d71d60effa to fa16c3d0bacd (3 revisions) (#62813)

Commit:
Increase the number of shards for web tests (#62623)

* increase the number of shards for web tests

* update comments

gaaclarke committed:
Added a more generous timeout period for the keyboard animation. (#62628)

a14n committed:
migrate part of painting to nullsafety (#62696)

* migrate part of painting to nullsafety

* address review comments

* address review comment

xster committed:
move compile/target sdk versions to 29 (#62799)

Commit:
[null-safety] remove some usages of mockito (#62809)

CareF committed:
separate scroll until visible implementation for lower level accessibility  (#62462)

CareF committed:
promote WidgetTester pumpAndSettle (#62640)

godofredoc committed:
Remove framework_tests from cirrus. (#62643)

rami-a committed:
Ensure dark theme is used when high contrast dark theme isn't provided (#62668)

engine-flutter-autoroll committed:
Roll Engine from fa16c3d0bacd to 033dd4560744 (3 revisions) (#62839)

goderbauer committed:
Revert "migrate part of painting to nullsafety (#62696)" (#62868)

This reverts commit 4518a72ffdc84ad950ae85749dcab02352e1ebc4.

Commit:
Revert "move compile/target sdk versions to 29 (#62799)" (#62884)

This reverts commit 5b3dc5251be6a56a8b65ae9e578ff2b330806ebc.

Commit:
plugin template  min flutter sdk version to 1.20 (#62605)

JuYeong0413 committed:
Upadte systemNavigationBarIconBrightness docs (#62729)

rami-a committed:
Address misc time picker design issues (#62803)

LongCatIsLooong committed:
longestLine layout width (#62657)

Hixie committed:
Enable a scrolling fling test. (#62825)

jason-simmons committed:
Fix text escaping in the AboutListTile sample code (#62707)

Hixie committed:
Update skipped test comment to point to an open bug (#62820)

Commit:
[null-safety] remove quiver from flutter tests (#62866)

Commit:
[null-safety] revert export of Fake from test_api (#62894)

goderbauer committed:
Reland "migrate part of painting to nullsafety (#62696)" (#62872)

darrenaustin committed:
TextSelectionTheme support (step 1 of 3) (#62014)

Create a new TextSelectionTheme for text selection properties.

Piinks reviewed (approved):
This LGTM! Thank you for the contribution! 🙂

Piinks commented:
It looks like updating your branch with the latest from master will resolve the failing checks here.

engine-flutter-autoroll committed:
Roll Engine from 033dd4560744 to 3081844cb4dd (9 revisions) (#62911)

Commit:
[null-safety] remove mockito from text control a11y test (#62869)

CareF committed:
revert widgettester breaking change (#62914)

Hixie committed:
Make the Skia expectation parser more resilient and update to "master_str". (#62898)

Piinks committed:
Add flutter_goldens_client to testing path (#62910)

* Add flutter_goldens_client for testing

* Update .cirrus.yml

CareF committed:
migrate macrobenchmark/multi_widget_construction test to e2e (#62539)

engine-flutter-autoroll committed:
Roll Engine from 3081844cb4dd to 4d5b0c31f8cb (10 revisions) (#62925)

Commit:
[flutter_tools] remove reference to flutter tool test from user generated gitignores (#62926)

engine-flutter-autoroll committed:
Roll Engine from 4d5b0c31f8cb to e1828008b97a (4 revisions) (#62938)

engine-flutter-autoroll committed:
a4e0104ae Roll Dart SDK from 987c9e347ca4 to 363c72111d12 (32 revisions) (flutter/engine#20251) (#62943)

Pragya007 committed:
Merge remote-tracking branch 'upstream/master' into RowtextDirection

Pragya007 commented:
@Piinks I have updated it.

Piinks commented:
Hey @Pragya007, sorry to ask again, but it looks like your last update might have picked up something else that is causing the devicelab tests to fail. I've re-run them a few times now to check. Can you update your branch once more?

Pragya007 commented:
@Piinks Sure!!

Pragya007 committed:
Merge branch 'master' into RowtextDirection

Pragya007 committed:
Merge branch 'RowtextDirection' of github.com:Pragya007/flutter into RowtextDirection
